app/businessrates/authorisation/action/AuthenticatedActionBuilder.scala

```diff
@@ -46,6 +49,16 @@ class AuthenticatedActionBuilder @Inject()(
           case (None, Some(_)) => Future.successful(Unauthorized(Json.obj("errorCode" -> "NO_EXTERNAL_ID")))
           case (None, None) => Future.successful(Unauthorized(Json.obj("errorCode" -> "NO_EXTERNAL_ID_OR_GROUP_ID")))
         }
+    }recoverWith {
+      case e: InternalError => {
+        logger.warn(e.getMessage)
+        throw e
+      }
+      case e: AuthorisationException => {
```

Comment:
is AuthorisationException the parent trait of all the errors we seen ?

Reply:
yeah - happy with this approach?

Reply:
Yes,